src/main/frontend/components/page.cljs

```diff
@@ -183,49 +183,63 @@
   (when title
     (let [*title-value (get state ::title-value)
           *edit? (get state ::edit?)
+          input-ref (rum/create-ref)
           repo (state/get-current-repo)
           hls-file? (pdf-assets/hls-file? title)
           title (if hls-file?
                   (pdf-assets/human-hls-filename-display title)
                   (if fmt-journal? (date/journal-title->custom-format title) title))
           old-name (or title page-name)
           confirm-fn (fn []
-                       (let [merge? (and (not= (util/page-name-sanity-lc page-name)
+                       (let [new-page-name (string/trim @*title-value)
+                             merge? (and (not= (util/page-name-sanity-lc page-name)
                                                (util/page-name-sanity-lc @*title-value))
                                          (page-handler/page-exists? page-name)
                                          (page-handler/page-exists? @*title-value))]
                          (ui/make-confirm-modal
                           {:title         (if merge?
-                                            (str "Page \"" @*title-value "\" already exists, merge them?")
-                                            "Do you really want to change the page name?")
+                                            (str "Page “" @*title-value "” already exists, merge to it?")
+                                            (str "Do you really want to change the page name to “" new-page-name "”?"))
                            :on-confirm    (fn [_e {:keys [close-fn]}]
                                             (close-fn)
                                             (page-handler/rename! (or title page-name) @*title-value)
                                             (reset! *edit? false))
                            :on-cancel     (fn []
                                             (reset! *title-value old-name)
+                                            (gobj/set (rum/deref input-ref) "value" old-name)
                                             (reset! *edit? true))})))
+          rollback-fn #(do
+                         (reset! *title-value old-name)
+                         (gobj/set (rum/deref input-ref) "value" old-name)
+                         (reset! *edit? false)
+                         (notification/show! "Illegal page name, can not rename!" :warning))
           blur-fn (fn [e]
+                    (when (util/wrapped-by-quotes? @*title-value)
```

Comment:
Is it good to change what users input? What if the user needs a title wrapped by quotes?

Reply:
This conflicts with `title::` property with double-quotes.

Reply:
hmm, I see. Definitely need some rules here as we talked about before.